### PR TITLE
LLVM: Move aliases to aliases.nix and switch to llvmPackages_latest

### PR DESCRIPTION
**Pull request by primeos**
###### Motivation for this change

See https://github.com/NixOS/nixpkgs/pull/116736#issuecomment-808227014.

Unfortunately `clang` and `llvm` (s. "TODO" in the diff) are used to widely so moving those attributes requires someone who knows how to do mass renames (via `rnix-parser`, etc.).

E.g. this is still very inconsistent (mixing `llvmPackages` and `llvmPackages_latest`):
```nix
  clang = llvmPackages.clang;
  clang-manpages = llvmPackages.clang-manpages;
  clang-tools = callPackage ../development/tools/clang-tools {
    llvmPackages = llvmPackages_latest;
  };
  clang-analyzer = callPackage ../development/tools/analysis/clang-analyzer {
    llvmPackages = llvmPackages_latest;
    inherit (llvmPackages_latest) clang;
  };
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -16776,6 +16783,8 @@ in
 
     inherit (pkgs.darwin) libobjc;
     inherit (pkgs.darwin.apple_sdk.frameworks) ApplicationServices OpenGL Cocoa AGL;
+
+    inherit (pkgs.llvmPackages) libcxx;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:16787]:
```suggestion
    inherit (pkgs.darwin) libobjc;
    inherit (pkgs.darwin.apple_sdk.frameworks) ApplicationServices OpenGL Cocoa AGL;
    inherit (pkgs.llvmPackages) libcxx;
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12393,9 +12403,6 @@ in
     inherit (llvmPackages_11) llvm clang-unwrapped;
   };
 
-  libcxx = llvmPackages.libcxx;
-  libcxxabi = llvmPackages.libcxxabi;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:12397]: Those should be moved to aliases.nix

Comment (primeos) [reply pkgs/top-level/all-packages.nix:12397]: Yes, this is already the case (/ purpose of this PR).

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -2536,7 +2538,9 @@ in
 
   grim = callPackage ../tools/graphics/grim { };
 
-  gringo = callPackage ../tools/misc/gringo { };
+  gringo = callPackage ../tools/misc/gringo {
+    inherit (pkgs.llvmPackages) libcxx;
```
[the remark this answers is not in the document]
Comment (primeos) [reply pkgs/top-level/all-packages.nix:2542]: Yes, I considered that but I'm not sure if it's worth it as `discord/base.nix` is called indirectly through `discord/default.nix`.

**Comment by primeos** [comment]
I'll drop this, not sure anymore if this is even a good idea (it would avoid some issues but likely also introduce new ones like making it harder to discover these aliases).

**Pull request closed without merging**